src/blockchain/compact_filters/peer.rs

```diff
@@ -508,6 +538,10 @@ impl InvPeer for Peer {
     }
 
     fn ask_for_mempool(&self) -> Result<(), CompactFiltersError> {
+        if !self.version.services.has(ServiceFlags::BLOOM) {
```

Comment:
Why asking the mempool needs bloom filter enabled?

Reply:
Not sure, but with `debug=net` i noticed this log line using Core 0.21:

```
```

Reply:
I think the `mempool` message was added as part of BIP37 and was meant to be called after a bloom filter was applied. But in the docs it doesn't say that you are not allowed to call it without filters, so I thought we could use it here